### ISSUE TITLE
Remove deprecated NetstringReceiver parser state attributes

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6321]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#6321|
|Type|task|
|Created|2013-02-23 16:00:40Z|
|Branch|https://github.com/twisted/twisted/tree/parser-attr-6321|

`LENGTH`, `DATA`, `COMMA`, and `NUMBER` in [twisted/protocols/basic.py](twisted/protocols/basic.py) are deprecated since 10.2 ([#4541](https://github.com/twisted/twisted/issues/4541)/r29962) and can be removed.

Attachments:

* [first-twisted-patch.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/aa4/aa4d19b925f7ca4f968ed09e72aaad7989dd4e2a/4771184fd129fd1f427949f74e29b6c5905e05c6.patch) (2423 bytes) - added by vasuman on 2013-04-16 10:33:26Z - 
* [6321.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/aa4/aa4d19b925f7ca4f968ed09e72aaad7989dd4e2a/1c337aaea2c322184edb3da987ac91834c4c0b33.patch) (638 bytes) - added by introom on 2013-04-16 14:07:29Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6321 6321
type__task task
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_parser_attr_6321 branches/parser-attr-6321
branch_author__vasuman vasuman
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1361635240000000 1361635240000000
changetime__1366291054000000 1366291054000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="introom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|introom set owner to introom<br>introom set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="introom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|introom removed owner<br>introom set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="introom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|introom commented|
|-|-|

only deleted "LENGTH, DATA, COMMA NUMBER" without modifying the corresponding test case as it's already deleted in ticket [#6042](https://github.com/twisted/twisted/issues/6042)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra set owner to introom|
|-|-|

Replying to [introom](#note_4):
> only deleted "LENGTH, DATA, COMMA NUMBER" without modifying the corresponding test case as it's already deleted in ticket [[[#6042](https://github.com/twisted/twisted/issues/6042)](https://github.com/twisted/twisted/issues/6042)](https://github.com/twisted/twisted/issues/6042)

I think you forgot to add the review keyword for this ticket? You should keep this ticket focused on one task and I don't see why #6042 should have the corresponding test case deleted, and not this ticket. It would mean merging this patch results in a broken test since the test case is there but the code's gone. Can you supply a new patch and keep everything related to this ticket here and not partly in #6042? Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

I now see vasuman also added a patch 2 days ago ([[first-twisted-patch.patch])](None/first-twisted-patch.patch])) which does everything we want. Not sure why introom added a new patch, but vasuman should've simply unassigned him or herself from the ticket and add the 'review' keyword.

I'll put that patch in a branch and run it on the buildslaves.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra commented|
|-|-|

(In [38137]) Branching to 'parser-attr-6321'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

(In [38138]) apply first-twisted-patch.patch, add news file. refs [#6321](https://github.com/twisted/twisted/issues/6321)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra commented|
|-|-|

Started a [build](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/parser-attr-6321), will merge when all's good.

**Comment by twisted-trac** [comment]
|<img alt="introom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_10"></a>|introom commented|
|-|-|

Replying to [thijs](#note_6):
> I now see vasuman also added a patch 2 days ago ([[first-twisted-patch.patch])](None/first-twisted-patch.patch])) which does everything we want. Not sure why introom added a new patch, but vasuman should've simply unassigned him or herself from the ticket and add the 'review' keyword.
> 
> I'll put that patch in a branch and run it on the buildslaves.

because I was handling ticket [#6042](https://github.com/twisted/twisted/issues/6042) which also modifies twisted.protocols.test.test_basic, so I modified it with consideration of this ticket. As this may cause some merging conflicts, so I planned to not add the review keyword (until 6042 is fixed).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [38148]) Merge parser-attr-6321: The NetstringReceiver parser state attributes in t.protocols.basic, deprecated since 10.2, are removed now.

Author: vasuman
Reviewer: thijs
Fixes: [#6321](https://github.com/twisted/twisted/issues/6321)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun commented|
|-|-|

Please follow the style guide at http://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles for news fragments.  Thanks.
